pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java

```diff
@@ -87,6 +90,14 @@ public class ConnectionPool implements AutoCloseable {
     /** Async release useless connections task. **/
     private ScheduledFuture asyncReleaseUselessConnectionsTask;
 
+
+    @Value
+    private static class Key {
+        private final InetSocketAddress logicalAddress;
+        private final InetSocketAddress physicalAddress;
+        private final int randomKey;
```

Comment:
Nit: private and final are redundant in the context of `Value` here.

```suggestion
        InetSocketAddress logicalAddress;
        InetSocketAddress physicalAddress;
        int randomKey;
```